src/lib/m2/index.js

```diff
@@ -116,5 +128,29 @@ export default new r.Struct({
   }),
   unknown2: new r.Optional(r.uint32le, function() {
     return this.flags === 8;
-  })
+  }),
+
+  isAnimated: function() {
+    let animated = false;
+
+    this.bones.forEach((bone) => {
+      if (bone.isAnimated) {
+        animated = true;
+      }
+    });
+
+    this.transparencies.forEach((transparency) => {
+      if (transparency.isAnimated) {
+        animated = true;
+      }
+    });
+
+    this.colors.forEach((color) => {
+      if (color.color.isAnimated || color.alpha.isAnimated) {
+        animated = true;
+      }
+    });
+
+    return animated;
+  }
```

Comment:
Ah that's a good point about `forEach`!

[`Array#some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) might help:

```
let animated = this.bones.some(bone => bone.isAnimated);
if (animated) {
  return true;
}
...
```

Might be premature optimization to do that.